File: frontend/src/types/Product.ts
import type { Category } from './Category';
import type { Brand } from './Brand';
import type { ProductVariant } from './ProductVariant';
import type { ProductImage } from './ProductImage';
import type { Review } from './Review';

export interface Product {
  id: number;
  name: string;
  slug: string;
  description: string;
  categoryId: number;
  brandId: number;
  status: string;
  createdAt: string;
  updatedAt: string;
  category?: Category;
  brand?: Brand;
  productVariant?: ProductVariant;
  images?: ProductImage[];
  reviews?: Review[];
}

File: backend/src/main/java/iuh/fit/backend/model/Product.java
package iuh.fit.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;
    private String slug;
    private String description;

    @Column(name = "category_id")
    private Integer categoryId;

    @Column(name = "brand_id")
    private Integer brandId;

    private String status;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "brand_id", insertable = false, updatable = false)
    private Brand brand;

    @OneToOne(mappedBy = "product", fetch = FetchType.LAZY)
    @JsonIgnore
    private ProductVariant productVariant;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<ProductImage> images;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<Review> reviews;
}